src/benchmarks/micro/coreclr/BilinearInterpol/BilinearInterpol.cs

```diff
@@ -0,0 +1,315 @@
+            Vector128<int> One = Vector128.Create(1);
+
+            for (var i = 0; i < x.Length; i += Vector256<double>.Count)
+            {
+                Vector256<double> currentX = Avx.LoadVector256(pX + i);
+
+                // Determine the largest a, such that A[i] = f(xA) and xA <= x[i].
+                // This involves casting from double to int; here we use a Vector conversion.
+                Vector256<double> aDouble = Avx.Multiply(Avx.Subtract(currentX, vMinXA), vInvDeltaA);
+                Vector128<int> a = Avx.ConvertToVector128Int32(aDouble);
+                a = Sse41.Min(Sse41.Max(a, Vector128<int>.Zero), ALengthMinusOne);
+                Vector128<int> aPlusOne = Sse41.Min(Sse2.Add(a, One), ALengthMinusOne);
+
+                // Now, get the reference input, xA, for our index a.
+                // This involves casting from  int to double.
+                Vector256<double> xA = Avx.Add(Avx.Multiply(Avx.ConvertToVector256Double(a), vDeltaA), vMinXA);
+
+                // Now, compute the lambda for our A reference point.
+                Vector256<double> currentXNormA = Avx.Max(vMinXA, Avx.Min(currentX, vMaxXA));
+                Vector256<double> lambdaA = Avx.Multiply(Avx.Subtract(currentXNormA, xA), vInvDeltaA);
+
+                // Now, we need to load up our reference points using Vector Gather operations.
+                Vector256<double> AVector = Avx2.GatherVector256(pA, a, 8);
+                Vector256<double> AVectorPlusOne = Avx2.GatherVector256(pA, aPlusOne, 8);
+
+                // Now, do the all of the above for our B reference point.
+                Vector256<double> bDouble = Avx.Multiply(Avx.Subtract(currentX, vMinXB), vInvDeltaB);
+                Vector128<int> b = Avx.ConvertToVector128Int32(bDouble);
+                b = Sse41.Min(Sse41.Max(b, Vector128<int>.Zero), BLengthMinusOne);
+                Vector128<int> bPlusOne = Sse41.Min(Sse2.Add(b, One), BLengthMinusOne);
+
+                Vector256<double> xB = Avx.Add(Avx.Multiply(Avx.ConvertToVector256Double(b), vDeltaB), vMinXB);
+                Vector256<double> currentXNormB = Avx.Max(vMinXB, Avx.Min(currentX, vMaxXB));
+                Vector256<double> lambdaB = Avx.Multiply(Avx.Subtract(currentXNormB, xB), vInvDeltaB);
+
+                Vector256<double> BVector = Avx2.GatherVector256(pB, b, 8);
+                Vector256<double> BVectorPlusOne = Avx2.GatherVector256(pB, bPlusOne, 8);
+
+                Vector256<double> newZ = Avx.Add(Avx.Multiply(vWeightA, Avx.Add(AVector, Avx.Multiply(lambdaA, Avx.Subtract(AVectorPlusOne, AVector)))),
+                                             Avx.Multiply(vWeightB, Avx.Add(BVector, Avx.Multiply(lambdaB, Avx.Subtract(BVectorPlusOne, BVector)))));
+                Avx.Store(pZ, newZ);
+            }
+        }
+        return z;
+    }
+#endif // !NETCOREAPP2_1 && !NETCOREAPP2_2 && !NETFRAMEWORK
+
+    public static bool CheckResult(double[] output, double[] vectorOutput)
```

Comment:
nit: is this a dead code that should be removed?

Reply:
Sort of; I'll add a comment. This was used to check that the vectorized versions gave the same result, but I couldn't figure out a good way to do that without including it in the benchmark timing. I'd prefer to leave it in so that, when adding new versions in future (e.g. perhaps an SSE* version and an Arm64 Intrinsics version), I can use this to make sure I've done it correctly.